### PR TITLE
Generate warning when a scrub is paused

### PR DESCRIPTION
**Pull request by themylogin**
Please not that alert logic itself is not implemented yet

### CODE REVIEW

**Comment by william-gr** [comment]
@themylogin Have you checked if a devd event is sent when a scrub is paused? I think that would be the best solution here. We can ask OS team to generate one if there isn't.

**Comment by william-gr** [comment]
Looks like it is:

```!system=ZFS subsystem=ZFS type=misc.fs.zfs.scrub_paused  pool_name=freenas-boot pool_guid=15592400354646457819```

**Comment by themylogin** [comment]
@william-gr I would prefer to get this as a state rather than as event. Is there any way to do that?

**Comment by william-gr** [comment]
@themylogin I will fix to get the state but polling for scrub paused is not a good idea in my opinion. Why dont you want an event?

**Comment by themylogin** [comment]
@william-gr Because alert checkers have polling model.
Also it will only poll once at 3AM because of the nature of this alert. We don't want alert immediately when scrub is paused, we want it if we forgot to unpause it overnight.

**Comment by william-gr** [comment]
@themylogin I see, I thought an immediate alert was asked. I will implement it in py-libzfs and let you know.

File: src/middlewared/middlewared/alert/source/scrub_paused.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+from middlewared.alert.base import Alert, AlertLevel, ThreadedAlertSource
+from middlewared.alert.schedule import CrontabSchedule
+
+
+class ScrubPausedAlertSource(ThreadedAlertSource):
+    level = AlertLevel.WARNING
+    title = "Scrub is paused"
+
+    schedule = CrontabSchedule(hour=13)
```
Comment (william-gr) [inline src/middlewared/middlewared/alert/source/scrub_paused.py]: Why hour 13?

Comment (themylogin) [reply src/middlewared/middlewared/alert/source/scrub_paused.py]: It was 13 when I was testing it :) I'll update it to 3 as soon as py-libzfs update will be available

**Comment by william-gr** [comment]
@themylogin https://github.com/freenas/py-libzfs/commit/3e4b10ac5f0acf296b3f988292560f522bbc342d

scrub.pause will be the `datetime` the scrub paused, None if not paused

**Comment by themylogin** [comment]
@william-gr done

**Pull request merged**